apps/jobs/views.py

```diff
@@ -276,14 +276,31 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
                 "request": request,
             },
         )
+        message = dict()
```

Comment:
Please make this corresponding change in the admin action for re-running a submission.

Reply:
I changes are reflected here: https://github.com/Cloud-CV/EvalAI/pull/2376/files#diff-eee3bef376ba38e2ff3289821573aa45R72-R76
Please correct me if I misunderstood this review

Reply:
Ohh..my bad. I didn't see the braces. Can you please create a variable there and pass it in the function.

Reply:
Hi,
Do you mean like this(from https://github.com/Cloud-CV/EvalAI/pull/2376/files#diff-42c752186f5f6b2ed96b92afa242a6f3R279-R302), pointed the changes using comments:
```py
######### CHANGE 1 <--
 message = {                                                
                "challenge_pk": challenge_id,
                "phase_pk": challenge_phase_id
        }
        if challenge.is_docker_based:
            try:
                file_content = json.loads(
                    request.FILES['input_file'].read()
                )
                message["submitted_image_uri"] = file_content["submitted_image_uri"]
            except:
                response_data = {
                    "error": "Error in deserializing submitted_image_uri from submission file"
                }
                return Response(
                    response_data, status=status.HTTP_400_BAD_REQUEST
                )
        if serializer.is_valid():
            serializer.save()
            response_data = serializer.data
            submission = serializer.instance
            ###########  CHANGE 2 <--
            message["submission_pk"]: submission.id
            # publish message in the queue
            publish_submission_message(
                challenge_id, challenge_phase_id, submission.id
            )
            publish_submission_message(message)
```

Reply:
No No, the change here is fine but in apps/jobs/admin.py file we will re-run the code upload challenge submission from admin then it will break because it won't have `submitted_image_uri` as a key.
So, I was suggesting to create a message dict there and then append the keys to it based upon the challenge type instead of directly passing the dict into the function.

I hope this is clear, right? Feel free to ask if you've any doubt.